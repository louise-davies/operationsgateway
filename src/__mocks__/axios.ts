// TODO: move __mocks__ folder back to package root once facebook/create-react-app#7539 is fixed

const requests = {
  get: jest.fn((path) => {
    if (path === '/operationsgateway-settings.json') {
      return Promise.resolve({
        data: {
          apiUrl: 'api',
        },
      });
    } else {
      return Promise.resolve({
        data: {},
      });
    }
  }),
  post: jest.fn(() => Promise.resolve({ data: {} })),
  delete: jest.fn(() => Promise.resolve({ data: {} })),
};

export default requests;
