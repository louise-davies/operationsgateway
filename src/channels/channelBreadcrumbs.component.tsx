import { NavigateNext } from '@mui/icons-material';
import { Breadcrumbs, Link, Typography } from '@mui/material';
import React from 'react';

type ChannelBreadcrumbsProps = {
  currNode: string;
  setCurrNode: (newNode: string) => void;
};

const ChannelBreadcrumbs = (props: ChannelBreadcrumbsProps) => {
  const { currNode, setCurrNode } = props;
  // special handling for '/' otherwise we get ['','']
  const nodePath = currNode === '/' ? [''] : currNode.split('/');

  return (
    <Breadcrumbs
      separator={<NavigateNext fontSize="small" />}
      aria-label="breadcrumb"
    >
      {nodePath.map((value, currIndex) => {
        const label = value.length ? value : 'All Channels';
        // final node, render as non-"link"
        if (currIndex === nodePath.length - 1) {
          return (
            <Typography key={label} color="text.primary">
              {label}
            </Typography>
          );
        } else {
          return (
            <Link
              key={label}
              href="#"
              onClick={(ev) => {
                // prevent default to stop the href="#" adding a hash to the URL
                // and potentially jumping the page
                ev.preventDefault();
                setCurrNode(nodePath.slice(0, currIndex + 1).join('/'));
              }}
            >
              {label}
            </Link>
          );
        }
      })}
    </Breadcrumbs>
  );
};

export default ChannelBreadcrumbs;
