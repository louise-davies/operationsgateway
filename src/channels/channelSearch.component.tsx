import { Search } from '@mui/icons-material';
import {
  Autocomplete,
  type FilterOptionsState,
  InputAdornment,
  TextField,
} from '@mui/material';
import React from 'react';
import { FullChannelMetadata } from '../app.types';

type ChannelSearchProps = {
  currPathAndChannel: string;
  channels: FullChannelMetadata[];
  onSearchChange: (channel: FullChannelMetadata) => void;
};

const filterOptions = (
  options: FullChannelMetadata[],
  { inputValue }: FilterOptionsState<FullChannelMetadata>
) => {
  const searchString = inputValue.toLowerCase();

  return options.filter((option) => {
    const name = option?.name?.toLowerCase();
    const systemName = option.systemName.toLowerCase();
    // need to search both friendly and system names for a match
    return name?.includes(searchString) || systemName.includes(searchString);
  });
};

const ChannelSearch = (props: ChannelSearchProps) => {
  const { onSearchChange, channels, currPathAndChannel } = props;

  const [value, setValue] = React.useState<FullChannelMetadata | null>(null);
  const [inputValue, setInputValue] = React.useState('');

  React.useEffect(() => {
    // clear the search box if the user clicks another channel / navigates via breadcrumbs
    if (value && currPathAndChannel !== `${value.path}/${value.systemName}`) {
      setValue(null);
      setInputValue('');
    }
  }, [currPathAndChannel, value]);

  return (
    <Autocomplete
      fullWidth
      filterOptions={filterOptions}
      value={value}
      inputValue={inputValue}
      onInputChange={(event, newInputValue) => setInputValue(newInputValue)}
      size="small"
      options={channels}
      getOptionLabel={(option) =>
        option.name
          ? `${option.name} (${option.systemName})`
          : option.systemName
      }
      blurOnSelect
      onChange={(event: unknown, newValue: FullChannelMetadata | null) => {
        if (newValue) onSearchChange(newValue);
        setValue(newValue);
      }}
      renderInput={(params) => (
        <TextField
          {...params}
          label="Search data channels"
          InputProps={{
            ...params.InputProps,
            startAdornment: (
              <InputAdornment position="start">
                <Search />
              </InputAdornment>
            ),
          }}
        />
      )}
    />
  );
};

export default ChannelSearch;
