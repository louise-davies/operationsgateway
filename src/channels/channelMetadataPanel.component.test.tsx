import { screen } from '@testing-library/react';
import React from 'react';
import ChannelMetadataPanel from './channelMetadataPanel.component';
import { FullChannelMetadata } from '../app.types';
import { PreloadedState } from '@reduxjs/toolkit';
import { QueryClient } from '@tanstack/react-query';
import { renderComponentWithProviders } from '../setupTests';
import { RootState } from '../state/store';
import axios from 'axios';
import { ChannelSummary, staticChannels } from '../api/channels';

describe('Channel Metadata Panel', () => {
  let displayedChannel: FullChannelMetadata | undefined;
  let channelSummary: ChannelSummary;

  const createView = (
    initialState?: PreloadedState<RootState>,
    queryClient?: QueryClient
  ) => {
    return renderComponentWithProviders(
      <ChannelMetadataPanel displayedChannel={displayedChannel} />,
      {
        preloadedState: initialState,
        queryClient,
      }
    );
  };

  beforeEach(() => {
    (axios.get as jest.Mock).mockResolvedValue({
      data: channelSummary,
    });

    channelSummary = {
      first_date: '2022-01-31T00:00:00',
      most_recent_date: '2023-01-31T00:00:00',
      recent_sample: [
        { '2022-01-31T00:00:00': 1 },
        { '2022-01-30T00:00:00': 2 },
        { '2022-01-29T00:00:00': 3 },
      ],
    };
  });

  it('should render correctly for no channel selected', () => {
    displayedChannel = undefined;
    const view = createView();

    expect(view.asFragment()).toMatchSnapshot();
  });

  it('should render correctly for system channel', async () => {
    displayedChannel = staticChannels['shotnum'];
    const view = createView();

    expect(view.asFragment()).toMatchSnapshot();
  });

  it('should render correctly for scalar channel with units selected', async () => {
    displayedChannel = {
      name: 'Test',
      systemName: 'test',
      type: 'scalar',
      units: 'cm',
      path: '/test',
      description: 'Test description',
    };
    const view = createView();
    await screen.findByText('Data Summary');

    expect(view.asFragment()).toMatchSnapshot();
  });

  it('should render correctly for waveform channel with units selected', async () => {
    displayedChannel = {
      systemName: 'test_2',
      type: 'waveform',
      x_units: 'mm',
      y_units: 'J',
      path: '/test_2',
      historical: true,
    };
    const view = createView();
    await screen.findByText('Data Summary');

    expect(view.asFragment()).toMatchSnapshot();
  });
});
