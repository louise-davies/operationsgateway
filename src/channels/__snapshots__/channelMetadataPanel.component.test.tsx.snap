// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Channel Metadata Panel > should render correctly for no channel selected 1`] = `
<DocumentFragment>
  <h3
    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-um3j9i-MuiTypography-root"
  >
    Data Channels Help
  </h3>
  <p
    class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
  >
    There are two options to locate a channel. You can either search or use the navigation on the left.
  </p>
  <p
    class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
  >
    Channel metadata can also be viewed. This can be displayed by clicking a channel name
  </p>
</DocumentFragment>
`;

exports[`Channel Metadata Panel > should render correctly for scalar channel with units selected 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-3y1137-MuiStack-root"
  >
    <h3
      class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
    >
      Channel_ABCDE
    </h3>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-11qr2p8-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z"
          />
        </svg>
      </span>
      Add this channel
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <p
    class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
  >
    System name: CHANNEL_ABCDE
  </p>
  <p
    class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
  >
    Test description
  </p>
  <p
    class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
  >
    Channel type: scalar
  </p>
  <p
    class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
  >
    Units: cm
  </p>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
  />
  <h4
    class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-w08jgs-MuiTypography-root"
  >
    Data Summary
  </h4>
  <p
    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
  >
    First data date: 2022-01-29 00:00:00
  </p>
  <p
    class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
  >
    Most recent data date: 2023-01-31 00:00:00
  </p>
  <h5
    class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-w08jgs-MuiTypography-root"
  >
    Recent Data
  </h5>
  <div
    class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
  >
    <table
      aria-label="recent data"
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
            scope="col"
          >
            Time
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
            scope="col"
          >
            Data
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            2022-01-31 00:00:00
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
          >
            6
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            2022-01-30 00:00:00
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
          >
            5
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            2022-01-29 00:00:00
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
          >
            4
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Channel Metadata Panel > should render correctly for system channel 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-3y1137-MuiStack-root"
  >
    <h3
      class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
    >
      Shot Number
    </h3>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-11qr2p8-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z"
          />
        </svg>
      </span>
      Add this channel
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <p
    class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
  >
    System name: shotnum
  </p>
  <p
    class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
  >
    Channel type: scalar
  </p>
</DocumentFragment>
`;

exports[`Channel Metadata Panel > should render correctly for waveform channel with units selected 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-3y1137-MuiStack-root"
  >
    <h3
      class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
    >
      CHANNEL_CDEFG
    </h3>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-11qr2p8-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z"
          />
        </svg>
      </span>
      Add this channel
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <p
    class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
  >
    Channel type: waveform
  </p>
  <p
    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
  >
    X Units: mm
  </p>
  <p
    class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
  >
    Y Units: J
  </p>
  <p
    class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-3784aq-MuiTypography-root"
  >
    This channel is historical
  </p>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
  />
  <h4
    class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-w08jgs-MuiTypography-root"
  >
    Data Summary
  </h4>
  <p
    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
  >
    First data date: 2022-01-29 00:00:00
  </p>
  <p
    class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
  >
    Most recent data date: 2023-01-31 00:00:00
  </p>
  <h5
    class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-w08jgs-MuiTypography-root"
  >
    Recent Data
  </h5>
  <div
    class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
  >
    <table
      aria-label="recent data"
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
            scope="col"
          >
            Time
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
            scope="col"
          >
            Data
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            2022-01-31 00:00:00
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
          >
            <img
              alt="CHANNEL_CDEFG waveform at 2022-01-31 00:00:00"
              src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/4gIoSUNDX1BST0ZJTEUAAQEAAAIYAAAAAAQwAABtbnRyUkdCIFhZWiAAAAAAAAAAAAAAAABhY3NwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAA9tYAAQAAAADTLQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAlkZXNjAAAA8AAAAHRyWFlaAAABZAAAABRnWFlaAAABeAAAABRiWFlaAAABjAAAABRyVFJDAAABoAAAAChnVFJDAAABoAAAAChiVFJDAAABoAAAACh3dHB0AAAByAAAABRjcHJ0AAAB3AAAADxtbHVjAAAAAAAAAAEAAAAMZW5VUwAAAFgAAAAcAHMAUgBHAEIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFhZWiAAAAAAAABvogAAOPUAAAOQWFlaIAAAAAAAAGKZAAC3hQAAGNpYWVogAAAAAAAAJKAAAA+EAAC2z3BhcmEAAAAAAAQAAAACZmYAAPKnAAANWQAAE9AAAApbAAAAAAAAAABYWVogAAAAAAAA9tYAAQAAAADTLW1sdWMAAAAAAAAAAQAAAAxlblVTAAAAIAAAABwARwBvAG8AZwBsAGUAIABJAG4AYwAuACAAMgAwADEANv/bAEMAAwICAwICAwMDAwQDAwQFCAUFBAQFCgcHBggMCgwMCwoLCw0OEhANDhEOCwsQFhARExQVFRUMDxcYFhQYEhQVFP/bAEMBAwQEBQQFCQUFCRQNCw0UFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFP/AABEIAB4AOAMBIgACEQEDEQH/xAAbAAABBQEBAAAAAAAAAAAAAAACAAMEBQYBCf/EADQQAAIBAgQEAggFBQAAAAAAAAECAwQRAAUSIRMxQVEGkRQiIzJScZKhFlNhgdIVQmKi0f/EABcBAQEBAQAAAAAAAAAAAAAAAAADAQL/xAAcEQEAAgMBAQEAAAAAAAAAAAABAAIRIUExYRL/2gAMAwEAAhEDEQA/APSHK1pIa6CndaZovRonMJWElBw2JNgNf9t7nbtjQU1BRysyyZdTxOAG08NTseXTnsdsUoEqzZVfiPBJFCNK62AAR9V1C23G3O+/LF5NXRpU076Zt9UZ9g/Ii9+XdR54mqPyUAT7DbK6BNRajpgqi5JiXb7YZpqGjmYrJl1PE2kOoMam6nvtse46bb47WV0bpwtM3tSqW4D7i/rdPhBwU9dGk9PJpm5lCTA+wI+XcLgrnXIAxvsMZVQaSTR0wAvvwl/5iNHTUTRuzZbTodHFQFF9ZfLY8rje1xh2rro2pjHpm9q3DtwHvYne23a5/bAZhVI6oypNc6oyeC42YWty+LThZR1yKhjfYGaZVQ/0irPoVODwHO0S/Cf0wsVuZ5Rl8lHVzcTM1kaNmGqrqlS9ibWLabXPLlyFrWwsUk4OV75vQkryp4QCVHLhP11k/wCo+ZxpKxC9M+kamWzqo6kG4H2xi8oz7L4liqyatpXhjGnhQ2Sy22NtXU8yeeJtd4xX0ZjRmc1A91Z1QIf0JG4xiZMTRw5mjDrUVsZU6kSPiXHc3Cny1Ydq42kp3CC7j1lHdgbj7gYxWV+Lp4Kw+kU6R0pux4LamLdNiBt+/QYuvxvQ/lVH0r/LHJXSPZ022JyWySLVTwFDqRVMt/ncL5gt5YeqozLTuq+/a636MNx97Yz8XjDLYNXDp511HUbKu5+rDn43ofyqj6V/lgV0/rsNtmOSVmkq1WWuU90UzzX7XQgA/O58sLFPX+LstTLa1Y6edS8T3OldzY/5YWFRPZlkfJ//2Q=="
              style="border: 1px solid #000000;"
            />
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            2022-01-30 00:00:00
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
          >
            <img
              alt="CHANNEL_CDEFG waveform at 2022-01-30 00:00:00"
              src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/4gIoSUNDX1BST0ZJTEUAAQEAAAIYAAAAAAQwAABtbnRyUkdCIFhZWiAAAAAAAAAAAAAAAABhY3NwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAA9tYAAQAAAADTLQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAlkZXNjAAAA8AAAAHRyWFlaAAABZAAAABRnWFlaAAABeAAAABRiWFlaAAABjAAAABRyVFJDAAABoAAAAChnVFJDAAABoAAAAChiVFJDAAABoAAAACh3dHB0AAAByAAAABRjcHJ0AAAB3AAAADxtbHVjAAAAAAAAAAEAAAAMZW5VUwAAAFgAAAAcAHMAUgBHAEIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFhZWiAAAAAAAABvogAAOPUAAAOQWFlaIAAAAAAAAGKZAAC3hQAAGNpYWVogAAAAAAAAJKAAAA+EAAC2z3BhcmEAAAAAAAQAAAACZmYAAPKnAAANWQAAE9AAAApbAAAAAAAAAABYWVogAAAAAAAA9tYAAQAAAADTLW1sdWMAAAAAAAAAAQAAAAxlblVTAAAAIAAAABwARwBvAG8AZwBsAGUAIABJAG4AYwAuACAAMgAwADEANv/bAEMAAwICAwICAwMDAwQDAwQFCAUFBAQFCgcHBggMCgwMCwoLCw0OEhANDhEOCwsQFhARExQVFRUMDxcYFhQYEhQVFP/bAEMBAwQEBQQFCQUFCRQNCw0UFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFP/AABEIAB4AOAMBIgACEQEDEQH/xAAbAAABBQEBAAAAAAAAAAAAAAACAAMEBQYBCf/EADQQAAIBAgQEAggFBQAAAAAAAAECAwQRAAUSIRMxQVEGkRQiIzJScZKhFlNhgdIVQmKi0f/EABcBAQEBAQAAAAAAAAAAAAAAAAADAQL/xAAcEQEAAgMBAQEAAAAAAAAAAAABAAIRIUExYRL/2gAMAwEAAhEDEQA/APSHK1pIa6CndaZovRonMJWElBw2JNgNf9t7nbtjQU1BRysyyZdTxOAG08NTseXTnsdsUoEqzZVfiPBJFCNK62AAR9V1C23G3O+/LF5NXRpU076Zt9UZ9g/Ii9+XdR54mqPyUAT7DbK6BNRajpgqi5JiXb7YZpqGjmYrJl1PE2kOoMam6nvtse46bb47WV0bpwtM3tSqW4D7i/rdPhBwU9dGk9PJpm5lCTA+wI+XcLgrnXIAxvsMZVQaSTR0wAvvwl/5iNHTUTRuzZbTodHFQFF9ZfLY8rje1xh2rro2pjHpm9q3DtwHvYne23a5/bAZhVI6oypNc6oyeC42YWty+LThZR1yKhjfYGaZVQ/0irPoVODwHO0S/Cf0wsVuZ5Rl8lHVzcTM1kaNmGqrqlS9ibWLabXPLlyFrWwsUk4OV75vQkryp4QCVHLhP11k/wCo+ZxpKxC9M+kamWzqo6kG4H2xi8oz7L4liqyatpXhjGnhQ2Sy22NtXU8yeeJtd4xX0ZjRmc1A91Z1QIf0JG4xiZMTRw5mjDrUVsZU6kSPiXHc3Cny1Ydq42kp3CC7j1lHdgbj7gYxWV+Lp4Kw+kU6R0pux4LamLdNiBt+/QYuvxvQ/lVH0r/LHJXSPZ022JyWySLVTwFDqRVMt/ncL5gt5YeqozLTuq+/a636MNx97Yz8XjDLYNXDp511HUbKu5+rDn43ofyqj6V/lgV0/rsNtmOSVmkq1WWuU90UzzX7XQgA/O58sLFPX+LstTLa1Y6edS8T3OldzY/5YWFRPZlkfJ//2Q=="
              style="border: 1px solid #000000;"
            />
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            2022-01-29 00:00:00
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
          >
            <img
              alt="CHANNEL_CDEFG waveform at 2022-01-29 00:00:00"
              src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/4gIoSUNDX1BST0ZJTEUAAQEAAAIYAAAAAAQwAABtbnRyUkdCIFhZWiAAAAAAAAAAAAAAAABhY3NwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAA9tYAAQAAAADTLQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAlkZXNjAAAA8AAAAHRyWFlaAAABZAAAABRnWFlaAAABeAAAABRiWFlaAAABjAAAABRyVFJDAAABoAAAAChnVFJDAAABoAAAAChiVFJDAAABoAAAACh3dHB0AAAByAAAABRjcHJ0AAAB3AAAADxtbHVjAAAAAAAAAAEAAAAMZW5VUwAAAFgAAAAcAHMAUgBHAEIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFhZWiAAAAAAAABvogAAOPUAAAOQWFlaIAAAAAAAAGKZAAC3hQAAGNpYWVogAAAAAAAAJKAAAA+EAAC2z3BhcmEAAAAAAAQAAAACZmYAAPKnAAANWQAAE9AAAApbAAAAAAAAAABYWVogAAAAAAAA9tYAAQAAAADTLW1sdWMAAAAAAAAAAQAAAAxlblVTAAAAIAAAABwARwBvAG8AZwBsAGUAIABJAG4AYwAuACAAMgAwADEANv/bAEMAAwICAwICAwMDAwQDAwQFCAUFBAQFCgcHBggMCgwMCwoLCw0OEhANDhEOCwsQFhARExQVFRUMDxcYFhQYEhQVFP/bAEMBAwQEBQQFCQUFCRQNCw0UFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFP/AABEIAB4AOAMBIgACEQEDEQH/xAAbAAABBQEBAAAAAAAAAAAAAAACAAMEBQYBCf/EADQQAAIBAgQEAggFBQAAAAAAAAECAwQRAAUSIRMxQVEGkRQiIzJScZKhFlNhgdIVQmKi0f/EABcBAQEBAQAAAAAAAAAAAAAAAAADAQL/xAAcEQEAAgMBAQEAAAAAAAAAAAABAAIRIUExYRL/2gAMAwEAAhEDEQA/APSHK1pIa6CndaZovRonMJWElBw2JNgNf9t7nbtjQU1BRysyyZdTxOAG08NTseXTnsdsUoEqzZVfiPBJFCNK62AAR9V1C23G3O+/LF5NXRpU076Zt9UZ9g/Ii9+XdR54mqPyUAT7DbK6BNRajpgqi5JiXb7YZpqGjmYrJl1PE2kOoMam6nvtse46bb47WV0bpwtM3tSqW4D7i/rdPhBwU9dGk9PJpm5lCTA+wI+XcLgrnXIAxvsMZVQaSTR0wAvvwl/5iNHTUTRuzZbTodHFQFF9ZfLY8rje1xh2rro2pjHpm9q3DtwHvYne23a5/bAZhVI6oypNc6oyeC42YWty+LThZR1yKhjfYGaZVQ/0irPoVODwHO0S/Cf0wsVuZ5Rl8lHVzcTM1kaNmGqrqlS9ibWLabXPLlyFrWwsUk4OV75vQkryp4QCVHLhP11k/wCo+ZxpKxC9M+kamWzqo6kG4H2xi8oz7L4liqyatpXhjGnhQ2Sy22NtXU8yeeJtd4xX0ZjRmc1A91Z1QIf0JG4xiZMTRw5mjDrUVsZU6kSPiXHc3Cny1Ydq42kp3CC7j1lHdgbj7gYxWV+Lp4Kw+kU6R0pux4LamLdNiBt+/QYuvxvQ/lVH0r/LHJXSPZ022JyWySLVTwFDqRVMt/ncL5gt5YeqozLTuq+/a636MNx97Yz8XjDLYNXDp511HUbKu5+rDn43ofyqj6V/lgV0/rsNtmOSVmkq1WWuU90UzzX7XQgA/O58sLFPX+LstTLa1Y6edS8T3OldzY/5YWFRPZlkfJ//2Q=="
              style="border: 1px solid #000000;"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
